accepted/targeting-packs-and-runtime-packs.md

```diff
@@ -0,0 +1,286 @@
+
+Now imagine this graph:
+
+* A (net45):
+   * `<PackageReference Include="mscorlib" Version="4.5.0">`
+   * `<PackageReference Include="System" Version="4.5.0">`
+   * `<ProjectReference Include="B.csproj" />`
+
+* B (net40):
+   * `<PackageReference Include="mscorlib" Version="4.0.0">`
+   * `<PackageReference Include="System" Version="4.0.0">`
+   * `<PackageReference Include="System.xml" Version="4.0.0">`
+
+* Applying standard nuget transitivity, A gets references:
+  * B
+  * mscorlib, 4.5.0
+  * System, 4.5.0
+  * System.Xml, 4.0.0 <-- Oops!
+
+In this example, for A to use B, System.Xml is needed, but it makes no sense to use .NET 4.0 System.Xml in .NET 4.5. So A must unify to .NET 4.5 System.Xml. In the case of System and mscorlib, this happened because A also references them directly. This breaks down when a framework asset is acquired transitively.
+
+
+## FAQ: If NuGet writes FrameworkReferences to assets file, shouldn't it also be responsible for determining FrameworkReference assets and listing them in assets file?
+
+Writing FrameworkReferences to the assets file is morally equivalent to writing References as frameworkAssemblies. The process of consuming them will be nearly the same: instead of raising Reference items from the assets file, we raise FrameworkReference. NuGet does not need to know anything about them other than their names.
+
+Abstractly, FrameworkReferences are not directly coupled to packages. As outlined above, they will commonly be resolved without any packages being used. The assets are only downloaded in packages to cover the scenario of building for a downlevel TFM where you don't have the targeting packs installed globally.
+
+This allows the SDK to be fully in control of how FrameworkReferences are resolved to files on disk without baking more concepts into NuGet.
+
+
+# Global resolution
+
+There is one weakness of historical framework NuGet packages that was listed in the introduction, but not adressed in the plan above: "Requires project context for resolution." The issue there is that reference assemblies may only be pulled down by a NuGet restore operation, which is still tied to a project. In general, you cannot just ask for the reference assemblies for a given TFM / shared framework unless that targeting pack is globally installed. Furthermore, Visual Studio has a Global Design Time Assembly Resolution (GDTAR) service that relies on being able to do just that. Platforms based on packages such as UWP and .NET Core cannot provide this service naturally, which blocks certain VS features from working correctly and there is concern that some of the features that will need to be brought up for .NET Core 3. WPF/WinForms may run into this. For UWP, design-time scenarios, this was an issue and a workaround that is not considered maintainable was instituted. For .NET Core 3, we will need to find a better solution. Investigation on that is tracked by https://github.com/dotnet/cli/issues/10094
```

Comment:
Not sure if this the same as what you are talking about here, but we'd like the ability to install reference assemblies for older frameworks in the .NET Core SDK docker images. We'd like to install 2.1 ref assemblies in 3.0 images and 2.1 and 3.0 ref assemblies in 3.1 SDK docker images. Is that satisfied by the new NuGet download only feature?

Reply:
It's not related to this paragraph.

Also, 2.x is out-of-scope for this feature. When targeting 2.x, it will use traditional nupkgs as always even if using the 3.0 SDK. The experience will be possible for > 3.0. For example. 3.1 SDK + 3.0 targeting pack can be deployed.

For 2.x, the offline story is nuget fallback folders. You can configure these in a docker image. For example, you could take the 2.x lzma and deploy it with the 3.0 SDK and 3.0 and 2.x targeting would work offline.

Reply:
Got it. That works. If we get significant customer feedback, we'll decide what to do.